src/components/common/Dialogs/AwardAndSmiteDialogs/ManageReputationContainer/ManageReputationContainer.tsx

```diff
@@ -0,0 +1,187 @@
+    defaultMessage: "Amount must be less than the user's reputation",
+  },
+});
+
+const defaultValidationSchema = object()
+  .shape({
+    domainId: number().required(),
+    user: object().shape({
+      walletAddress: string().address().required(),
+    }),
+    amount: string()
+      .required()
+      .test(
+        'more-than-zero',
+        () => MSG.amountZero,
+        (value) => {
+          const numberWithoutCommas = (value || '0').replace(/,/g, ''); // @TODO: Remove this once the fix for FormattedInputComponent value is introduced.
+          return !new Decimal(numberWithoutCommas).isZero();
+        },
+      ),
+    annotation: string().max(4000),
+    forceAction: boolean(),
+    motionDomainId: number(),
+  })
+  .defined();
+
+type FormValues = InferType<typeof defaultValidationSchema>;
+
+const ManageReputationContainer = ({
+  colony: { nativeToken, watchers },
+  colony,
+  callStep,
+  prevStep,
+  cancel,
+  close,
+  filteredDomainId,
+  isSmiteAction = false,
+  enabledExtensionData,
+}: AwardAndSmiteDialogProps) => {
+  const [isForce, setIsForce] = useState(false);
+  const [userReputation, setUserReputation] = useState(0);
+  const navigate = useNavigate();
+  const colonyWatchers =
+    watchers?.items.filter(notNull).map((item) => item.user) || [];
+
+  // const verifiedUsers = useMemo(() => {
+  //   return getVerifiedUsers(colony.whitelistedAddresses, colonyWatchers) || [];
+  // }, [colonyWatchers, colony]);
+
+  const updateReputationCallback = (userRepPercentage: number) => {
+    setUserReputation(userRepPercentage);
+  };
+
+  const { isVotingReputationEnabled } = enabledExtensionData;
+
+  const actionType =
+    !isForce && isVotingReputationEnabled
+      ? ActionTypes.MOTION_MANAGE_REPUTATION
+      : ActionTypes.ACTION_MANAGE_REPUTATION;
+
+  let smiteValidationSchema;
+
+  if (isSmiteAction) {
+    const amountValidationSchema = object()
+      .shape({
+        amount: string()
+          .required()
+          .test(
+            'more-than-zero',
+            () => MSG.amountZero,
+            (value) => {
+              const numberWithoutCommas = (value || '0').replace(/,/g, ''); // @TODO: Remove this once the fix for FormattedInputComponent value is introduced.
+              return !new Decimal(numberWithoutCommas).isZero();
+            },
+          )
+          .test(
+            'less-than-user-reputation',
+            () => MSG.maxAmount,
+            (value) => {
+              const numberWithoutCommas = (value || '0').replace(/,/g, ''); // @TODO: Remove this once the fix for FormattedInputComponent value is introduced.
+              return !new Decimal(numberWithoutCommas).greaterThan(
+                userReputation,
+              );
+            },
+          ),
+      })
+      .required();
+    smiteValidationSchema = defaultValidationSchema.concat(
+      amountValidationSchema,
+    );
+  }
+
+  const nativeTokenDecimals = nativeToken?.decimals || DEFAULT_TOKEN_DECIMALS;
```

Comment:
Could you use the `getTokenDecimalsWithFallback` method instead?

Reply:
True, I forgot about this function